macos/FluentUI/Button/Button.swift

```diff
@@ -121,6 +122,37 @@ open class Button: NSButton {
 		}
 	}
 
+	func setupBorderShadowsIfNeeded() {
+		// At the moment border shadows only apply for the Primary and Secondary Button Styles.
+		// Border shadows being meant for buttons and their corresponding States, that have ~some~
+		// background color other than `clear` which is why they don't apply to the Borderless Style
+		// buttons. Acrylic Style buttons do possess a background color, yet they maintain transparency.
+		// This transparency allows any underlying border shadow layers to bleed through, consequently
+		// impacting the overall button background color. Adding shadows in this context presents a
+		// challenge. Therefore, unless it becomes necessary, these buttons will remain without shadows.
+		// And perhaps shadowPath is the way to go, where you can have the CALayer have a Clear
+		// background color and still show a shadow by setting a hollow shadowPath around the button.
+		self.usesBorderShadows = style == .primary || style == .secondary
+	}
+
+	var firstOuterDropShadowLayer: CALayer?
+	var secondOuterDropShadowLayer: CALayer?
+	var innerShadowLayer: CALayer?
+
+	open override func layout() {
+		super.layout()
+		if usesBorderShadows {
+			// Set the frame of all the shadow layers to the same bounds as this Button View. This
+			// couldn't have been done in the init where the shadow layers are setup because the Button
+			// view doesn't yet have any bounds established, so we're forced to do it in the layout
+			// override where we're guaranteed to have bounds and is also invoked every time the view's
+			// size or position needs adjustments.
+			firstOuterDropShadowLayer?.frame = self.bounds
```

Comment:
should generally assign to a layer's bounds instead of frame since frame is a computed property. From the documentation, modifying frame could lead to side-effects by modifying the other properties which factor into it (position, anchorPoint)

Reply:
When I do this, the layer's positioning gets messed up. After a quick bit of research, it looks like frame gets the correct result because we're positioning the layer within the view (the layer's parent), so we're working in the parent's coordinate system. But using layer.bounds instead possibly is getting messy, because this could be modifying the layer's own coordinate system instead of its size and position in the parent's coordinate system. 

![Screenshot 2023-07-18 at 10 03 24](https://github.com/microsoft/fluentui-apple/assets/25355516/ff4b9496-a451-4ea5-82c3-a00207b3b777)

Reply:
All this is reminding me is I love views and autolayout :). But yeah going with bounds seems correct enough because we, in practice, are not modifying anchorPoint/position. I think I'd like to set aside some time to go deep dive into this stuff again. Is it possible the "right" thing to do is set frame, but we also need to convert from bounds from our layer's coordinate system to the sublayer's coordinate system?